icu4j/main/core/src/main/java/com/ibm/icu/util/LocaleData.java

```diff
@@ -225,7 +225,12 @@ public UnicodeSet getExemplarSet(int options, int extype) {
     public static final LocaleData getInstance(ULocale locale) {
         LocaleData ld = new LocaleData();
         ld.bundle = (ICUResourceBundle)UResourceBundle.getBundleInstance(ICUData.ICU_BASE_NAME, locale);
-        ld.langBundle = (ICUResourceBundle)UResourceBundle.getBundleInstance(ICUData.ICU_LANG_BASE_NAME, locale);
+        try {
+            ld.langBundle = (ICUResourceBundle)UResourceBundle.getBundleInstance(ICUData.ICU_LANG_BASE_NAME, locale);
+        } catch(Exception e) {
+            // Expected error case: ICU-22149
+            ld.langBundle = null;
```

Comment:
optional: The "cleaner" C++ solution of storing the UErrorCode, if you decide to do so, would have a parallel in Java in the form of storing the Exception in a new field.